### PR TITLE
Viem client type

### PR DESCRIPTION
**Pull request by agualis**
As a consumer, I want to have access to the type of the `client` parameter when calling methods like: 

`Relayer.signRelayerApproval` or `Permit2Helper.signAddLiquidityApproval` 

This PR refactors internal usages and exposes type:

```TS
export type ViemClient = Client & WalletActions & PublicActions;
```

### CODE REVIEW

File: src/utils/types.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { Client, WalletActions, PublicActions } from 'viem';
+
+export type ViemClient = Client & WalletActions & PublicActions;
```
Comment (agualis) [inline src/utils/types.ts:3]: We could be more explicit with the type name using something like: 

`type PublicWalletViemClient` or so but, given that it is the only client that we are exposing, I would keep it shorter.

File: test/entities/swaps/v2/swapV2.integration.test.ts [hunk 2]
```diff
@@ -44,7 +42,7 @@ const WETH = TOKENS[chainId].WETH;
 const vault = VAULT[chainId];
 
 describe('SwapV2', () => {
-    let client: Client & PublicActions & TestActions & WalletActions;
+    let client: ViemClient & TestActions;
```
Comment (MattPereira) [inline test/entities/swaps/v2/swapV2.integration.test.ts:45]: I guess my lean would be to go with a type name of  `PublicWalletClient` for situations like this

As you mentioned, we are only using Viem so i dont think having it as part of type name adds much meaning

**Review by MattPereira — APPROVED** [review]
I'm mostly indifferent here with the explicit vs DRY trade-off, but I do support any changes that are helpful for our most important consumer (the v3 FE)

But plz wait for @johngrantuk to comment in case I'm missing some reason we shouldn't do this

File: src/utils/types.ts (hunk 1 above)
Comment (agualis) [reply src/utils/types.ts]: Update: renamed to `PublicWalletClient`

**Comment by agualis** [comment]
> I'm mostly indifferent here with the explicit vs DRY trade-off, but I do support any changes that are helpful for our most important consumer (the v3 FE)

For more context: the important thing from the consumer POV is exposing the type, if not, we need to do some [TS acrobatics to satisfy the compiler](https://github.com/balancer/frontend-v3/blob/fixUndefinedTokensInRemoveProvider/lib/modules/relayer/signRelayerApproval.ts#L9).

File: src/utils/types.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { Client, WalletActions, PublicActions } from 'viem';
+
+export type PublicWalletClient = Client & WalletActions & PublicActions;
```
Comment (johngrantuk) [inline src/utils/types.ts:3]: Could we replace this with:
```
export type PublicWalletClient = WalletClient & PublicActions;
```
I believe that is a bit tidier and should work.

**Review by johngrantuk — APPROVED** [review]
Apart from the one little change I suggested it all looks good to me. Thanks.

**Pull request merged**